src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8620,28 +8621,36 @@ private static bool IsAccessibilityModifier(SyntaxKind kind)
             SyntaxListBuilder badBuilder = null;
             for (int i = 0; i < modifiers.Count; i++)
             {
-                switch (modifiers[i].ContextualKind)
+                var modifier = modifiers[i];
+                switch (modifier.ContextualKind)
                 {
                     case SyntaxKind.AsyncKeyword:
                         IsInAsync = true;
                         forceLocalFunc = true;
-                        break;
+                        continue;
                     case SyntaxKind.UnsafeKeyword:
                         forceLocalFunc = true;
-                        break;
-                    case SyntaxKind.StaticKeyword:
+                        continue;
                     case SyntaxKind.ReadOnlyKeyword:
                     case SyntaxKind.VolatileKeyword:
-                        break; // already reported earlier, no need to report again
-                    default:
-                        if (badBuilder == null)
+                        continue; // already reported earlier, no need to report again
+                    case SyntaxKind.StaticKeyword:
+                        if (IsFeatureEnabled(MessageID.IDS_FeatureStaticLocalFunctions))
```

Comment:
We should produce the same parsed tree (modulo diagnostics) regardless of language version.
Consider using `modifier = CheckFeatureAvailability(modifier, ...)` instead.